twoliter/embedded/Makefile.toml

```diff
@@ -1412,6 +1421,42 @@ pubsys \
 '''
 ]
 
+[tasks.fetch-variant]
+script_runner = "bash"
+script = [
+'''
+set -e
+
+export PATH="${TWOLITER_TOOLS_DIR}:${PATH}"
+
+if [ -z "${FETCH_VARIANT_LOCAL_ROOT_JSON}" ]; then
+   fetch_variant_root="./fetch-variant-root.json"
+   echo "Fetching root.json..."
+   curl -s -L -C - "${FETCH_VARIANT_ROOT_JSON}" -o "${fetch_variant_root#./}"
+   sha512sum -c <<<"${FETCH_VARIANT_ROOT_JSON_SHA512}  ${fetch_variant_root#./}"
+else
+   fetch_variant_root="${FETCH_VARIANT_LOCAL_ROOT_JSON}"
+fi
```

Comment:
These values need to come from `Infra.toml`.

If you have `FETCH_REPO` you should be able to use its `root_role_url` and `root_role_sha512` fields for this. `pubsys-setup` could probably be extended for this.